lib/semantics/resolve-names.cc

```diff
@@ -2138,17 +2214,76 @@ void DeclarationVisitor::SetType(
   symbol.SetType(type);
 }
 
-// ResolveNamesVisitor implementation
+// Find the Symbol for this derived type.
+const Symbol *DeclarationVisitor::ResolveDerivedType(const SourceName &name) {
```

Comment:
might be a function that could return a `std::optional<T>`

Reply:
It can't be `std::optional<Symbol>` because I don't want to copy Symbols, I need references to them. `std::optional<Symbol *>` isn't really an improvement.